### PR TITLE
Pass grid to target velocity in Compute*Quantities

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

I will need this quantity for size control. This just passes it, but it's currently not used. That will be a future PR.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by nilsdeppe — DISMISSED** [review]
LGTM but needs a rebase to resolve conflicts

**Comment by knelli2** [comment]
@nilsdeppe Rebased

**Comment by knelli2** [comment]
Maybe @nilsvu since Nils Deppe is away this week?

File: src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp [hunk 1]
```diff
@@ -668,7 +668,7 @@ void compute_dest_vars_from_source_vars(
       InterpolationTargetTag::compute_vars_to_interpolate::apply(
           dest_vars, source_vars, mesh, jac_grid_to_inertial,
           invjac_grid_to_inertial, jac_logical_to_grid, invjac_logical_to_grid,
-          inertial_mesh_velocity);
+          inertial_mesh_velocity, tnsr::I<DataVector, 3, Frame::Grid>{});
```
Comment (nilsvu) [inline src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp:671]: Please document this call signature. I see there's a protocol, so document it there.

Comment (knelli2) [reply src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp:671]: The protocol docs just point to the example, so I updated the example (which was actually incorrect but we never checked that the examples conformed to the protocol?? so I added some asserts as well)

Comment (nilsvu) [reply src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp:671]: I think the Example.hpp isn't actually compiled, so the asserts don't do anything.

Comment (knelli2) [reply src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp:671]: Ok added cmake stuff so it's compiled

File: tests/Unit/Helpers/ParallelAlgorithms/Interpolation/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+set(LIBRARY "ParallelInterpolationHelpers")
+
+add_spectre_library(${LIBRARY} INTERFACE)
+
+spectre_target_headers(
+  ${LIBRARY}
+  INCLUDE_DIRECTORY ${CMAKE_SOURCE_DIR}/tests/Unit
+  HEADERS
+  Examples.hpp
```
Comment (nilsvu) [inline tests/Unit/Helpers/ParallelAlgorithms/Interpolation/CMakeLists.txt]: This won't compile the file, you'd have to move it out of Helpers/ into a cpp file.

Comment (knelli2) [reply tests/Unit/Helpers/ParallelAlgorithms/Interpolation/CMakeLists.txt]: I get a build error though if I change something in this file though (like misspell a class name)

Comment (nilsvu) [reply tests/Unit/Helpers/ParallelAlgorithms/Interpolation/CMakeLists.txt]: Ah the file `tests/Unit/ParallelAlgorithms/Interpolation/Test_Protocols.cpp` is compiled and has the asserts.

**Comment by knelli2** [comment]
@nilsvu Did you want me to change anything else?

**Comment by nilsvu** [comment]
Yea the Examples.hpp thing isn't quite sorted out. Don't add the asserts in the hpp file. Overall, all I'm asking for is that the signature of the function you changed is documented.

**Comment by knelli2** [comment]
Ok I removed the asserts in the hpp file. And it is documented now because I changed Examples.hpp

File: src/ParallelAlgorithms/Interpolation/Protocols/ComputeVarsToInterpolate.hpp [hunk 3]
```diff
@@ -92,8 +92,14 @@ struct ComputeVarsToInterpolate {
...
                                         Frame::Inertial>&>(),
+            std::declval<const InverseJacobian<DataVector, Dim, Frame::Grid,
+                                               Frame::Inertial>&>(),
+            std::declval<const Jacobian<DataVector, Dim, Frame::ElementLogical,
+                                        Frame::Grid>&>(),
             std::declval<const InverseJacobian<
-                DataVector, Dim, Frame::ElementLogical, Frame::Grid>&>()))>>
+                DataVector, Dim, Frame::ElementLogical, Frame::Grid>&>(),
+            std::declval<const tnsr::I<DataVector, Dim, Frame::Inertial>&>(),
+            std::declval<tnsr::I<DataVector, Dim, Frame::Grid>&>()))>>
```
Comment (nilsvu) [inline src/ParallelAlgorithms/Interpolation/Protocols/ComputeVarsToInterpolate.hpp]: const ref

File: tests/Unit/Helpers/ParallelAlgorithms/Interpolation/Examples.hpp [hunk 4]
```diff
@@ -91,7 +91,8 @@ struct ExampleComputeVarsToInterpolate
   template <typename SrcTagList, typename DestTagList, size_t Dim>
   static void apply(
       const gsl::not_null<Variables<DestTagList>*> /*target_vars*/,
-      const Variables<SrcTagList>& /*src_vars*/, const Mesh<Dim>& /*mesh*/) {
+      const Variables<SrcTagList>& /*src_vars*/,
+      const Mesh<Dim - 1>& /*mesh*/) {
```
Comment (nilsvu) [inline tests/Unit/Helpers/ParallelAlgorithms/Interpolation/Examples.hpp]: This shouldn't be Dim - 1 I think?

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: tests/Unit/Helpers/ParallelAlgorithms/Interpolation/Examples.hpp (hunk 4 above)
Comment (knelli2) [reply tests/Unit/Helpers/ParallelAlgorithms/Interpolation/Examples.hpp]: Oops, yeah I was just testing out that the other signature was correct by making this one incorrect, but forgot to undo it.

**Review by nilsvu — APPROVED** [review]

**Pull request merged**